File: Kotlin/프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.kt
class Solution {
    fun solution(babbling: Array<String>): Int {
        var answer: Int = 0

        val ayayewooma = arrayOf("aya", "ye", "woo", "ma")
        val ayaayayeyewoowoomama = arrayOf("ayaaya", "yeye", "woowoo", "mama")
        val babblingCopy = babbling.copyOf() //"ayaye", "uuu", "yeye", "yemawoo", "ayaayaa"

        for (i in ayayewooma.indices) {
            val aya = ayayewooma[i]
            val ayaaya = ayaayayeyewoowoomama[i]

            for (j in babblingCopy.indices) {
                if (babblingCopy[j].contains(ayaaya)) {
                    babblingCopy[j] = babblingCopy[j].replace(ayaaya, "1")
                } else if (babblingCopy[j].contains(aya)) {
                    babblingCopy[j] = babblingCopy[j].replace(aya, " ")
                }
            }
        }

        return babblingCopy.count { it.isBlank() }
    }
}

File: C#/프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.cs
using System;

public class Solution {
    public int solution(string[] babbling) {
        int answer = 0;
        string[] bab = new string[] {"aya", "ye", "woo", "ma"};
        string[] bab_dou = new string[] {"ayaaya", "yeye", "woowoo", "mama"};
        
        for(int i = 0; i < babbling.Length; i++)
        {
            for(int j = 0; j < 4; j++)
            {
                if(babbling[i].Contains(bab_dou[j]))
                {
                    babbling[i] = babbling[i].Replace(bab_dou[j], " ");
                    continue;
                }
                
                if(babbling[i].Contains(bab[j]))
                {
                    babbling[i] = babbling[i].Replace(bab[j], j.ToString());
                }
            }
            
            string s = babbling[i];
            
            if(s.Contains(" ")) continue;
            if(int.TryParse(s, out int aasdsad) == false) continue;
            
            answer++;
        }
        
        return answer;
    }
}